### PR TITLE
chore: add docker compose configuration

### LINKED ISSUE #3: Configure initial docker image
### Description

To run the project we need an easy-to-transport infrastructure for the server where the web application will live. In this sense we decided to go for the use of docker containers to set up the infrastructure necessary for the project.

We will start with a monolithic architecture, i.e. a front-end consuming data from a back-end and two stores. The first is a database and the second is a document and image storage (see the desing).

- [ ] learn more about docker and docker compose
- [ ] add initial docker configuration to run the front-end, back-end, database and storage

---

### Desing

image

---

### References

- https://docs.docker.com/compose/
- https://www.drawio.com/

### PR DESCRIPTION
**Pull request by anghelo-code**
### Changes

Add Docker Compose configuration for two services:
- npm service: Installs all project dependencies using npm.
- frontend service: Runs the project in a development environment, making it easier to test and debug.

Close #3

### CODE REVIEW

**Comment by Kleiber** [comment]
We are going to create a new repository for the infrastructure so we can containerize our entire project. What do you think?

**Comment by anghelo-code** [comment]
I like the idea

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+services:
+  npm:
+    container_name: kratos-npm
+    image: node:22-alpine3.19
```
Comment (Kleiber) [inline docker/docker-compose.yml:4]: Instead of using this image directly, could we have our own image where we contain our services?

**Review by Kleiber — APPROVED** [review]

File: docker/docker-compose.yml (hunk 1 above)
Comment (Kleiber) [inline docker/docker-compose.yml:4]: Instead of using this image directly, could we have our own image where we contain our services?

**Review by Kleiber — CHANGES_REQUESTED** [review]

**Comment by Kleiber** [comment]
Not make sense for now since we will start to use google cloud services

**Pull request closed without merging**